diff -Naur tcp_wrappers_7.6/Makefile tcp_wrappers_7.6.b/Makefile
--- tcp_wrappers_7.6/Makefile	2012-06-09 12:40:44.445399978 -0500
+++ tcp_wrappers_7.6.b/Makefile	2012-06-09 13:08:19.049545514 -0500
@@ -149,9 +149,9 @@
 	EXTRA_CFLAGS="-DSYS_ERRLIST_DEFINED -DINET6 -Dss_family=__ss_family -Dss_len=__ss_len" VSYSLOG= all
 
 linux:
-	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS="-lnsl" RANLIB=ranlib ARFLAGS=rv AUX_OBJ=weak_symbols.o \
-	NETGROUP="-DNETGROUP" TLI= EXTRA_CFLAGS="$(RPM_OPT_FLAGS)  -fPIC -DPIC -D_REENTRANT -DSYS_ERRLIST_DEFINED -DBROKEN_SO_LINGER -DINET6=1 -Dss_family=__ss_family -Dss_len=__ss_len -DHAVE_WEAKSYMS" all
+	@$(MAKE) REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
+	LIBS="-lnsl" ARFLAGS=rc AUX_OBJ=weak_symbols.o \
+	NETGROUP="-DNETGROUP" TLI= EXTRA_CFLAGS="$(RPM_OPT_FLAGS)  -fPIC -DPIC -D_REENTRANT -DSYS_ERRLIST_DEFINED -DBROKEN_SO_LINGER -DHAVE_WEAKSYMS" all
 
 linux-old:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
@@ -395,13 +395,14 @@
 # library; some don't care as long as the modules are in the right order;
 # some systems don't even have a ranlib(1) command. Make your choice.
 
-RANLIB	= ranlib	# have ranlib (BSD-ish UNIX)
+#RANLIB	= ranlib	# have ranlib (BSD-ish UNIX)
 #RANLIB	= echo		# no ranlib (SYSV-ish UNIX)
 
-ARFLAGS	= rv		# most systems
+#ARFLAGS	= rv		# most systems
+ARFLAGS	= rc
 #ARFLAGS= rvs		# IRIX 4.0.x
 
-AR	= ar
+#AR	= ar
 #AR	= bld		# Unicos 8.x
 
 #############################################################################
@@ -572,7 +573,7 @@
 # still do selective username lookups as documented in the hosts_access.5
 # and hosts_options.5 manual pages (`nroff -man' format).
 #
-AUTH	= #-DALWAYS_RFC931
+AUTH	= -DALWAYS_RFC931
 #
 # The default username lookup timeout is 10 seconds. This may not be long
 # enough for slow hosts or networks, but is enough to irritate PC users.
@@ -680,10 +681,11 @@
 SHELL	= /bin/sh
 .c.o:;	$(CC) $(CFLAGS) -c $*.c
 
-CFLAGS	= -DFACILITY=$(FACILITY) $(ACCESS) $(PARANOID) $(NETGROUP) \
+CFLAGS	+= -DFACILITY=$(FACILITY) $(ACCESS) $(PARANOID) $(NETGROUP) \
 	$(BUGS) $(SYSTYPE) $(AUTH) $(UMASK) \
 	-DREAL_DAEMON_DIR=\"$(REAL_DAEMON_DIR)\" $(STYLE) $(KILL_OPT) \
 	-DSEVERITY=$(SEVERITY) -DRFC931_TIMEOUT=$(RFC931_TIMEOUT) \
+	-DPIC -D_REENTRANT -fPIC -DHAVE_STRERROR \
 	$(UCHAR) $(TABLES) $(STRINGS) $(TLI) $(EXTRA_CFLAGS) $(DOT) \
 	$(VSYSLOG) $(HOSTNAME)
 
@@ -726,7 +728,7 @@
 	-$(RANLIB) $(LIB)
 
 $(SHLIB): $(LIB_OBJ)
-	gcc -shared -fPIC -Wl,-z,relro -Wl,-soname -Wl,$(SHLIB).$(MAJOR) \
+	$(CC) $(CFLAGS) $(LDFLAGS) -shared -fPIC -Wl,-soname -Wl,$(SHLIB).$(MAJOR) \
 		-o $(SHLIB).$(MAJOR).$(MINOR).$(REL) $^ $(LIBS)
 	ln -s $(SHLIB).$(MAJOR).$(MINOR).$(REL) $(SHLIB).$(MAJOR)
 	ln -s $(SHLIB).$(MAJOR).$(MINOR).$(REL) $(SHLIB)
